### PR TITLE
Updating script to support Windows and python 3.8+

### PR DESCRIPTION
**Pull request by lmmfranco**
Main changes:
- Swapped `is` for `==` to resolve a syntax warning on python 3.8+ ([source](https://bugs.python.org/issue34850))
- Using `os.path.join` instead of manual string concatenation for paths to resolve script not working outside unix environments
- Added field initialization to `GoogleCredentials.__creds` to avoid crashing on current python
- Fixed code blocks on README.MD
- ~README: Added optional instructions for creating an alias on Windows~

### CODE REVIEW

**Review by psteiger — CHANGES_REQUESTED** [review]
Hi @lmmfranco , thanks a lot for your contribution! All the changes look good to me.

I have suggested some minor editing on the README file, please check.

Have you tested this script on Windows with your changes?

File: README.md
Comment (lmmfranco) [reply README.md:53]: I did not suggest adding the folder to `PATH` because `gdrive` is not a file that can be run from path, only executable files (.exe, .cmd, .bat, .ps1, etc) can be run this way.
Can we skip this suggestion?

**Comment by lmmfranco** [comment]
> I have suggested some minor editing on the README file, please check.
I left a reply there, but in short, I guess the suggestion does work on Windows as you'd expect

> Have you tested this script on Windows with your changes?
Yes. Both windows native command line interface and WSL (Windows Subsystem for Linux)

File: README.md
Comment (lmmfranco) [reply README.md:53]: > It does not smell good to suggest to manually alter System32 folder.

Oh, I missed that part. I can remove this section altogether if you prefer.

Comment (lmmfranco) [reply README.md:53]: An alternative would be to include a local `gdrive.bat` on this repo and add that to the `PATH`. I could do that

Comment (psteiger) [reply README.md:53]: Can't .py files be executed directly with the right shebang? 

https://www.python.org/dev/peps/pep-0397/#python-script-launching

https://docs.python.org/3/using/windows.html#python-launcher-for-windows

Comment (psteiger) [reply README.md:53]: See https://docs.python.org/3/faq/windows.html#how-do-i-make-python-scripts-executable

Comment (lmmfranco) [reply README.md:53]: As far as I know, this approach requires the source file to have `.py` extension for windows to resolve the shebang. `gdrive` has no extension

Comment (psteiger) [reply README.md:53]: I see. I'm willing to rename the file to `gdrive.py` and, to keep behavior on *nix, create a symbolic link named `gdrive` pointing to `gdrive.py`. For Windows, we can just run the `gdrive.py` directly, assuming the shebang logic would work, or create a `gdrive.bat` to launch the script by just running `gdrive` with the folder on PATH. Would that work for Windows?

Please test the behavior of renaming `gdrive` to `gdrive.py` and executing the file on Windows. If it works, I'll commit the rename and the symbolic link to this branch, and you can rebase your pull request.

Comment (lmmfranco) [reply README.md:53]: It seems windows will default to use pyhton2 on `.py` files when called directly, and only resolves shebangs when using `py` executable to run the file
image
(repeated lines stand for me trying out different shebangs)

Tests using `gdrive.py`:
image
(Syntax errors are caused by trying to run the script on python2)

Then finally adding a `gdrive.bat` containing this:
```bat
@echo off
py %~dp0\gdrive.py %*
```
(`%dp0` returns batch file directory, `%*` means all arguments)
And adding repository to the `PATH` env var, worked out fine:
image

Comment (psteiger) [reply README.md:53]: Thank you very much for your tests.

I have tested `gdrive.py` on Windows 10 with only a Python 3.9 installation (no Python 2). It worked fine. As Python 2 is EOL already, we should not be worrying very much about peculiarities in setups with both versions installed. Nevertheless, I think we should create a `gdrive.bat` as you described, so it will work nicely in your setup as well.

I have opened a pull request on #42 that renames the script and creates a symbolic link for *nix. Awaiting for review. Once merged, you can rebase your Pull Request.

Meanwhile, I suggest you add the following changes to your pull request:

1. Create `gdrive.bat` as you described.
2. Adjust README to reflect those new changes.

Thanks!

**Comment by lmmfranco** [comment]
Alright. Batch file created.
Steps for editing `System32` removed since `PATH` appending will now work out of the box.

Waiting for #42 to be merged before rebasing this patch.

**Comment by psteiger** [comment]
#42 has been merged.

**Comment by lmmfranco** [comment]
Rebased with master and added one additional change:
Listen for ctr+c (SIGINT/KeyboardInterrupt) on `list` command instead of outputing error

**Review by psteiger — APPROVED** [review]

**Comment by psteiger** [comment]
Please re-do the commits from HEAD of master. 

So we will not have commits such as "Reimplement changes lost in rebase", or empty commits. Then push force changes.

**Comment by lmmfranco** [comment]
Squashed all commits into a single commit after HEAD for better history

**Comment by psteiger** [comment]
@lmmfranco almost set. I only ask for one more change: fix commit message of the single comment. It should contain only:

- Add batch file for window execution

- Stop listing on SIGINT (KeyboardInterrupt)

**Comment by lmmfranco** [comment]
Done. Please review @psteiger

**Review by psteiger — APPROVED** [review]
Just one more thing -- just kidding. Looks great. Thanks @lmmfranco for your contribution and your patience. Approved. Go ahead and merge.

**Comment by lmmfranco** [comment]
> Just one more thing -- just kidding. Looks great.
Phew... 😅

> Go ahead and merge.
I'm afraid I don't have permission to do that. Only project members are allowed

**Pull request merged**
